common/network-yarn/pom.xml

```diff
@@ -174,7 +174,18 @@
             <configuration>
                 <target>
                     <echo message="Shade netty native libraries to ${spark.shade.native.packageName}" />
-                    <unzip src="${shuffle.jar}" dest="${project.build.directory}/exploded/" />
+                    <!-- 
+                      We have a LICENSE file with apache licenses and license directory for netty-tcnative licenses. This breaks
+                      packaging on OSX where filesystems are case insensitive. So, create the directory, which results
+                      in the LICENSE file not being written, letting the netty licenses be copied over.
+                      LICENSE.txt also contains the apache license so this should be OK 
+                    -->
+                    <mkdir dir="${project.build.directory}/exploded/META-INF/LICENSE" />
+                    <unzip src="${shuffle.jar}" dest="${project.build.directory}/exploded/">
+                      <patternset>
+                          <exclude name="META-INF/LICENSE"/>
+                      </patternset>
+                    </unzip>
```

Comment:
https://github.com/apache/spark/blob/ffc6994f46534dd955a387cea51afb9a67cb781e/common/network-yarn/pom.xml#L106-L114

How about adding `<exclude>META-INF/license/*</exclude>` directly to the above `excludes` list?

Reply:
trying that out now, will report back!